src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -1,12 +1,74 @@
+/* eslint-disable react/no-unused-state */
 import React from 'react';
+import { getUser } from '../../api/api';
 import './CurrentUser.scss';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+type Props = {
+  userId: number | null;
+  clear: () => void;
+};
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+type State = {
+  user: User | null;
+};
+
+export class CurrentUser extends React.PureComponent <Props, State> {
+  state = {
+    user: {
+      name: '',
+      id: 0,
+      email: '',
+      phone: '',
+    },
+  };
+
+  componentDidMount() {
+    getUser(this.props.userId)
+      .then((user) => {
+        this.setState({ user });
+      });
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.userId !== this.props.userId) {
+      getUser(this.props.userId)
+        .then((user) => {
+          this.setState({ user });
+        });
+    }
```

Comment:
This class extends PureComponent which is already have shallow comparison of props implemented in shouldComponentUpdate so this check is redundant